Develop customer add dual mode

fengtai-telink opened the pull request:
#### Testing
Verified by CI

fengtai-telink committed:
riscv: telink: fix matter CI

 - Fix the matter CI items .

Signed-off-by: Fengtai Xie <fengtai.xie@telink-semi.com>

haiwentelink commented on config/telink/app/bootloader.conf:
```diff
@@ -14,9 +14,9 @@
+# Enable this option in case if restoring the slot0 partition is expected from slot1
 # partition in case if slot0 is not bootable or damaged
-CONFIG_BOOT_BOOTSTRAP=y
+CONFIG_BOOT_BOOTSTRAP=n
```
should use DUAL_MODE define to distingwish normal mode and dual mode .

haiwentelink commented on config/telink/app/bootloader.conf:
```diff
@@ -27,7 +27,7 @@ CONFIG_BOOT_SWAP_USING_SCRATCH=n
 # Enable this option in case if the whole slot0 image need to be validated
 # With disabled option the only image magic is validated
-CONFIG_BOOT_VALIDATE_SLOT0=y
+CONFIG_BOOT_VALIDATE_SLOT0=n
```
should use DUAL_MODE define to distingwish normal mode and dual mode .

fengtai-telink committed:
riscv: telink: add the dual mode proc

 - Add the switch logic part .
 - Clean the boot flag in init part .

Signed-off-by: Fengtai Xie <fengtai.xie@telink-semi.com>

fengtai-telink committed:
riscv: telink: add dual-mode config

 - Add dual-mode config in build .
 - Add dual-mode CI items for b92(4mb), 2mb buteo and tercel .
 - Add partition for Zigbee binary(TELINK_ZB_PARTITION_ADDR) instead of hardcoded values .
 - Add independent bootloader for dual-mode .
 - Update zephyr hash .

Signed-off-by: Fengtai Xie <fengtai.xie@telink-semi.com>
